File: algorithm/lev3 네트워크.cpp
#include <string>
#include <vector>

using namespace std;

bool visit[201] = { false, };
int answer = 0;

void dfs(int idx, vector<vector<int>> c) {
	visit[idx] = true;
	for (int j = 0; j < c.size(); j++) {
		if (idx == j) continue;
		if (!visit[j] && c[idx][j]) {
			dfs(j, c);
		}
	}
}

int solution(int n, vector<vector<int>> computers) {
	for (int i = 0; i < n; i++) {
		if (!visit[i]) {
			dfs(i, computers);
			answer++;
		}
	}
	return answer;
}

File: algorithm/lev3 네트워크.py
def solution(n, computers):
    answer = 0
    cnt = 0
    visited = [0]*n
    def dfs(c):
        visited[c] = 1
        for s in range(n):
            if visited[s] is not 1 and computers[c][s] is 1:
                dfs(s)
    for i in computers:
        if visited[cnt] is not 1:
            dfs(cnt)
            answer+=1
        cnt+=1
    return answer
